xml_converter/src/xml_converter.cpp

```diff
@@ -217,80 +249,72 @@ void process_data(
 }
 
 ////////////////////////////////////////////////////////////////////////////////
-// main
-//
-// Main is the CLI entrypoint to the xml converter. It handles processing all
-// of the command line data into a format the internal functions want to
-// receive.
+// parse_arguments
+// Processes all of the command line data into a format the internal functions
+// want to receive.
 //
 // Example usage
-//   ./xml_converter --input-taco-paths ../packs/marker_pack --output-split-guildpoint-path ../output_packs
-//   ./xml_converter --input-taco-paths ../packs/* --output-split-guildpoint-path ../output_packs
+//   ./xml_converter --input-taco-paths ../packs/marker_pack --output-guildpoint-path ../output_packs
+//   ./xml_converter --input-taco-paths ../packs/* --output-guildpoint-path ../output_packs --split_by_map_id
+//
+// --import_FORMAT_paths <path> --FLAGS [--import_FORMAT_paths <path> --FLAGS]
+// --export_FORMAT_paths <path> --filter "conditions" --FLAGS
+// [--export_FORMAT_paths <path> --filter "conditions" --FLAGS]
 ////////////////////////////////////////////////////////////////////////////////
-int main(int argc, char* argv[]) {
-    vector<string> input_taco_paths;
-    vector<string> output_taco_paths;
-    vector<string> input_guildpoint_paths;
-    vector<string> output_guildpoint_paths;
-    bool allow_duplicates = false;
-
-    // Typically "~/.local/share/godot/app_userdata/Burrito/protobins" for
-    // converting from xml markerpacks to internal protobuf files.
-    vector<string> output_split_guildpoint_paths;
-
-    vector<string>* arg_target = nullptr;
+vector<PathConfig> parse_arguments(int argc, char* argv[]) {
```

Comment:
This function is complex enough that should have unit tests. Look at other examples of the C++ unit tests, like those for the string hierarchy class.

Reply:
Added tests